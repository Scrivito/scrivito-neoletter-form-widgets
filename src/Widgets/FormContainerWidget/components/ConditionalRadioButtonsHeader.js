import * as React from "react";
import * as Scrivito from "scrivito";
import { getFieldName } from "../utils/getFieldName";
import { SelectItem } from "./SelectComponent";
import { Mandatory } from "./MandatoryComponent";
import { HelpText } from "./HelpTextComponent";

export const RadioButtonsHeader = Scrivito.connect(
  ({ widget, onChangeSelected }) => {
    return (
      <div
        className={`mb-3 select-container ${
          Scrivito.isInPlaceEditingActive() ? "condition-wrapper" : ""
        }`}
      >
        <div className="select-title">
          <span className="text-super">{widget.get("title")}</span>
          {widget.get("required") && <Mandatory />}
          {widget.get("helpText") && <HelpText widget={widget} />}
        </div>
        <div className="row ">
          {widget.get("conditions").map((condition, index) => (
            <SelectItem
              selectionType={"radio"}
              name={getFieldName(widget)}
              onChange={onChangeSelected}
              value={condition.get("title")}
              id={condition.id()}
              required={widget.get("required")}
              key={index}
            />
          ))}
        </div>
      </div>
    );
  }
);
