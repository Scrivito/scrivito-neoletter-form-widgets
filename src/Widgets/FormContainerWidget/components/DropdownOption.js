import * as React from "react";

export const DropdownOption = ({ value, id }) => {
  return (
    <option value={value} id={id}>
      {value}
    </option>
  );
};
