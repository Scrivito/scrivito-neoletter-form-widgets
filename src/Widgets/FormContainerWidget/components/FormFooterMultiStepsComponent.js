import * as React from "react";
import * as Scrivito from "scrivito";

export const FormFooterMultiSteps = Scrivito.connect(
  ({
    widget,
    onPageChange,
    onSubmit,
    currentStep,
    isLastPage,
    stepsLength,
  }) => {
    return (
      <div className="form-buttons">
        <button
          className="btn btn-primary backward-button"
          onClick={() => onPageChange(false)}
          hidden={currentStep == 1 && !Scrivito.isInPlaceEditingActive()}
        >
          {widget.get("backwardButtonText")}
        </button>
        <div className="step-counter">{currentStep + " / " + stepsLength}</div>
        <button
          className="btn btn-primary forward-button"
          onClick={isLastPage ? onSubmit : () => onPageChange(true)}
        >
          {isLastPage
            ? widget.get("submitButtonText")
            : widget.get("forwardButtonText")}
        </button>
      </div>
    );
  }
);
