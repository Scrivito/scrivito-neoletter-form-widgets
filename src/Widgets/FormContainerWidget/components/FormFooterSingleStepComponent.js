import * as React from "react";
import * as Scrivito from "scrivito";

export const FormFooterSingleStep = Scrivito.connect(({ widget, onSubmit }) => {
  return (
    <div
      className={`${
        widget.get("singleSubmitButtonAlignment") === "block"
          ? ""
          : widget.get("singleSubmitButtonAlignment")
      }`}
    >
      <button
        className={`btn btn-primary${
          widget.get("singleSubmitButtonAlignment") === "block"
            ? " btn-block"
            : ""
        }`}
        onClick={onSubmit}
      >
        {widget.get("submitButtonText")}
      </button>
    </div>
  );
});
