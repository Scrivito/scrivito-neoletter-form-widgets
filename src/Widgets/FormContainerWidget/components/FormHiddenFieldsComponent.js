import * as React from "react";
import * as Scrivito from "scrivito";
import { getHistory } from "../../../config/history";

export const FormHiddenFields = ({ widget }) => {
  const [browserLocation, setBrowserLocation] = React.useState(null);
  React.useEffect(() => {
    const history = getHistory();
    console.log("And the history is....");
    console.log(history);
    if (!history) return;
    setBrowserLocation(locationToUrl(history.location));

    return history.listen(({ location }) =>
      setBrowserLocation(locationToUrl(location))
    );
  }, []);
  return (
    <>
      <input type="hidden" name="form_id" value={widget.get("formId")} />
      <input
        type="hidden"
        name="url"
        value={browserLocation || Scrivito.urlFor(widget.obj())}
      />
      <Scrivito.InPlaceEditingOff>
        <Scrivito.ContentTag content={widget} attribute="hiddenFields" />
      </Scrivito.InPlaceEditingOff>

      <HoneypotField />
    </>
  );
};

const HoneypotField = () => (
  <div aria-hidden="true" className="winnie-the-pooh">
    <input autoComplete="off" name="fax" tabIndex="-1" />
  </div>
);

function locationToUrl(location) {
  return `${window.location.origin}${location.pathname}${location.search}`;
}
