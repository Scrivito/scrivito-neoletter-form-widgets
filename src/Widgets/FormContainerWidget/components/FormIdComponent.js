import * as React from "react";
import * as Scrivito from "scrivito";
import { ContentProperty } from "../../../Components/ContentProperty";
import { getScrivitoFormWidgetConfig } from "../../../config/scrivitoConfig";

export const FormIdComponent = Scrivito.connect(({ widget }) => {
  const formSubmissionsHref = widget.get("formId")
    ? `https://edit.neoletter.com/i/${
        getScrivitoFormWidgetConfig().tenant
      }/forms/${widget.get("formId")}`
    : null;
  const uiContext = Scrivito.uiContext();
  if (!uiContext) return null;

  return (
    <div className={`scrivito_${uiContext.theme}`}>
      <div className="scrivito_detail_content">
        <div className="attribute_form_id_item">
          <ContentProperty
            content={widget}
            attribute="formId"
            title="Form ID"
            description="This ID identifies the form in Neoletter."
          />

          <a
            className={`scrivito_button ${
              formSubmissionsHref ? "scrivito_blue" : "scrivito_disabled"
            }`}
            href={formSubmissionsHref}
            target="_blank"
            rel="noreferrer"
          >
            View submissions
          </a>
        </div>
      </div>
    </div>
  );
});
