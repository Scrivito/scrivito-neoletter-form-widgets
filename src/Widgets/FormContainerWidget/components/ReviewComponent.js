import * as React from "react";
import Button from "react-bootstrap/Button";
import Modal from "react-bootstrap/Modal";

export const Review = ({ widget, reviewData, onHide }) => {
  const [show, setShow] = React.useState(false);

  React.useEffect(() => {
    setShow(true);
  }, [reviewData]);

  const handleClose = () => {
    setShow(false);
    onHide();
  };
  return (
    <Modal show={show} onHide={handleClose} size="lg">
      {widget.get("showReviewHeader") && (
        <Modal.Header style={{ alignSelf: "center" }}>
          <Modal.Title>{widget.get("reviewHeaderTitle")}</Modal.Title>
        </Modal.Header>
      )}
      <Modal.Body className="form-review">
        {reviewData.map((step, i) => (
          <div
            className="step-review-container"
            key={"step-review-container-" + i}
          >
            {widget.get("showStepsInReview") && (
              <h4 className="step-review">Step {i}</h4>
            )}
            {step.map((d, i) => (
              <ReviewItem key={"item-" + i} item={d} />
            ))}
          </div>
        ))}
      </Modal.Body>
      {widget.get("showReviewFooter") && (
        <Modal.Footer>
          <Button variant="primary" onClick={handleClose}>
            {widget.get("reviewCloseButtonText")}
          </Button>
        </Modal.Footer>
      )}
    </Modal>
  );
};

const ReviewItem = ({ item }) => {
  return (
    <div>
      <div className="review-item-title">
        <span>{item.title}</span>
      </div>
      <div className="review-item-value">{item.value}</div>
    </div>
  );
};
