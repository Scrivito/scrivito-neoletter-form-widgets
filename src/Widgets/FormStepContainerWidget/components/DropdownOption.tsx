import * as React from "react";
interface DropdownOptionProps {
  value: string;
  id: string;
}
export const DropdownOption: React.FC<DropdownOptionProps> = ({
  value,
  id
}) => {
  return (
    <option value={value} id={id}>
      {value}
    </option>
  );
};
