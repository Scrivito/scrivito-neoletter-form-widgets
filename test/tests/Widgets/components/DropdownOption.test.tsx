import React from "react";
import { render } from "@testing-library/react";
import { DropdownOption } from "../../../../src/Widgets/FormStepContainerWidget/components/DropdownOption"; // Update the path according to your actual component structure
import renderer from "react-test-renderer";

describe("DropdownOption", () => {
  const props = {
    value: "Apple",
    id: "apple-option"
  };

  it("renders with correct value and id", () => {
    const { container } = render(<DropdownOption {...props} />);
    const optionElement = container.querySelector("option");

    expect(optionElement).toBeInTheDocument();
    expect(optionElement).toHaveValue("Apple");
    expect(optionElement).toHaveAttribute("id", "apple-option");
    expect(optionElement).toHaveTextContent("Apple");
  });

  it("renders with different value and id", () => {
    const differentProps = {
      value: "Banana",
      id: "banana-option"
    };

    const { container } = render(<DropdownOption {...differentProps} />);
    const optionElement = container.querySelector("option");

    expect(optionElement).toBeInTheDocument();
    expect(optionElement).toHaveValue("Banana");
    expect(optionElement).toHaveAttribute("id", "banana-option");
    expect(optionElement).toHaveTextContent("Banana");
  });

  it("renders correctly", () => {
    const tree = renderer.create(<DropdownOption {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
