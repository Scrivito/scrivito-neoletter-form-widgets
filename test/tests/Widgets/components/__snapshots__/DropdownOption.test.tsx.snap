// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownOption renders correctly 1`] = `
<option
  id="apple-option"
  value="Apple"
>
  Apple
</option>
`;
